cmd/agent/main.go

```diff
@@ -3,6 +3,8 @@ package main
 import (
 	"os"
 
+	// This package should be initialized before 'k8s.io/apiserver/pkg/server', so that
+	// the internal registry could be used by controller-runtime.
```

Comment:
Hi, I didn't get why this should be initialized before 'k8s.io/apiserver/pkg/server'? Can you remind me?

Reply:
Hi @RainbowMango, this is under cover.

See [here](https://github.com/karmada-io/karmada/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/metrics/workqueue.go#L99-L100). The `workqueueMetricsProvider` will be registered into `globalMetricsFactory`. However, the provider will be only set once according to [here](https://github.com/karmada-io/karmada/blob/master/vendor/k8s.io/client-go/util/workqueue/metrics.go#L223-L228). If `k8s.io/apiserver/pkg/server` is initialized first, there is no chance to set the provider in `sigs.k8s.io/controller-runtime/pkg/metrics` because k8s workqueue metrics will be set first. See [here](https://github.com/karmada-io/karmada/blob/master/vendor/k8s.io/component-base/metrics/prometheus/workqueue/metrics.go#L101-L102)

Reply:
I think it's worth adding more clear comments on it. Just like you explained here.
It's very tricky and brokes the normal `imports-group` rule.